DeepLinkDemo/DeepLinkDemo/Controllers/HomeViewController.swift

```diff
@@ -104,13 +119,84 @@ class HomeViewController: UITableViewController {
+                                
+                                if let jsonObject = try JSONSerialization.jsonObject(with: data, options : .allowFragments) as? [String:AnyObject]
+                                {
+                                    if ((jsonObject["dma_eea"] != nil) && (jsonObject["dma_eea"] != nil) && (jsonObject["dma_eea"] != nil)) {
+                                        let dma_eea = (jsonObject["dma_eea"] as! String).lowercased() == "yes" ? true : false
+                                        let dma_ad_personalization = (jsonObject["dma_ad_personalization"] as! String).lowercased() == "yes" ? true : false
+                                        let dma_ad_user_data = (jsonObject["dma_ad_user_data"] as! String).lowercased() == "yes" ? true : false
+                                        self.logData = ""
+                                        self.enableBranchLogging(){(msg:String,msg2:BranchLogLevel,msg3:Error?)->() in
+                                            if (msg.contains("BranchSDK")){
+                                                self.logData = self.logData + msg + "\n"
+                                            }
+                                            vc?.updateText(msg: self.logData)
+                                        }
+                                        if(self.branchSDKInitialized){
+                                            Branch.getInstance().resetUserSession()
+                                        }
+                                        
+                                        Branch.setDMAParamsForEEA(dma_eea, adPersonalizationConsent: dma_ad_personalization, adUserDataUsageConsent: dma_ad_user_data)
+                                        AppDelegate.shared.getBranchData(AppDelegate.shared.launchOption)
+                                        self.branchSDKInitialized = true
+                                    } else {
+                                        self.logData = "Missing params from JSON Object: \n" + jsonObject.description
+                                    }
+                                } else {
+                                    self.logData = "Bad JSON : \n" + ProcessInfo.processInfo.arguments[i]
+                                }
+                            }
+
+                            
+                        }
+                    } catch let error as NSError {
+                        print(error)
+                        self.logData += error.localizedDescription
+                    }
+                    vc?.updateText(msg: self.logData)
+                    self.navigationController?.pushViewController(vc!, animated: true)
+                    
+               } else if textValue == "sendV2Event" {
+                   self.logData = ""
+                   
+                   let storyBoard : UIStoryboard = UIStoryboard(name: "Main", bundle:nil)
```

Comment:
Similar request to move check logic into it's own method.

Reply:
Done.

Reply:
Done